### PR TITLE
refactor: consolidate text-width helpers onto shared primitives

### PR DESCRIPTION
**Pull request by robbiew**
## Summary

Removes the root cause behind the 17 byte-vs-rune bugs fixed in #148 and #149: the repo had **six truncation helpers, four pad helpers and three `centerText`s**, each re-deriving the same logic, so every new screen re-invented one and had an even chance of measuring bytes.

They now delegate to shared primitives in `internal/ansi`:

```go
func TruncateRunes(s string, maxRunes int, ellipsis string) string
func PadRight(s string, width int) string
func PadLeft(s string, width int) string
func Center(s string, width int) string
```

## This is de-duplication, not unification

The variants differ deliberately, and those differences are user-visible column budgets:

| Helper | Ellipsis | Trims? |
| --- | --- | --- |
| `menu.truncateStr` | `..` | no |
| `menu.truncateString` | `...` | no |
| `menu.truncateRunes` | `...` | yes |
| `menu.adminTruncate` | `…` (1 rune, 3 bytes) | yes |
| `usereditor.truncateRunes` | none | no |

Each keeps its own convention and becomes a thin wrapper passing its ellipsis to the primitive. Unifying them would shift column budgets across many screens — a deliberate follow-up, not a rider on this change.

One rule generalizes all five:

```
if RuneCount(s) <= maxRunes          -> s
e := RuneCount(ellipsis)
if maxRunes <= e                     -> hard cut, no ellipsis
else                                 -> runes[:maxRunes-e] + ellipsis
```

Verified by hand against each variant before writing it — including `adminTruncate`'s `…`, which is one rune but three bytes, so the boundary had to be rune-based to match its old `max <= 1`.

## Proof that nothing changed

Characterization tests were written **first, in their own commit** (`10cac58`), pinning each helper's current output across no-truncation, exact-boundary, overflow, every hard-cut branch, empty input, whitespace-only, and multi-byte cases.

The migration commit (`aff8751`) touches **only production files** — no test was modified to make the migration pass. That's the evidence the output is identical.

## One behavioural divergence, documented rather than buried

For a **negative** `maxLen`/`width`, the old hand-rolled helpers panicked on the slice bound; the primitives clamp to 0 and return gracefully. Every current call site passes a positive literal or a pre-clamped value, so this is unreachable today, and graceful is the better behaviour — but it is a genuine difference and worth stating.

## Also fixed

`internal/usereditor/view.go`'s `centerText` was still byte-based — my #149 fix covered `fields.go` in the same package and missed this one. Every caller passes ASCII, so it's unreachable today; converted anyway, with the commit saying so plainly.

## Out of scope, deliberately

Unifying the ellipsis conventions; reconciling `configeditor`'s non-truncating `centerText` with `menueditor`'s truncating one; wide-character/CJK double-width support (would need `go-runewidth` threaded through `ansi.VisibleLength`); and `ansi.TruncateVisible`/`VisibleLength`, which were already correct and ANSI-aware.

## Verification

- `go test ./...` exit 0; `go test -race -count=1` across ansi, menu, usereditor, menueditor, configeditor exit 0
- `gofmt -l` empty; `go vet ./...` clean; `staticcheck` zero findings; binary builds

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved Unicode-safe text truncation, padding, and centering across menus and editors, avoiding cutting multi-byte characters incorrectly.
  * Standardized truncation/ellipsis behavior and edge-case handling for empty, whitespace-only, narrow, and zero-width values.
* **Tests**
  * Expanded unit and coverage for empty inputs, exact-fit vs truncation, hard-cut behavior, ellipsis variations, whitespace preservation, and UTF-8/rune boundary correctness.

### CODE REVIEW

File: internal/menu/executor_oneliners_text.go [hunk 1]
```diff
@@ -3,21 +3,18 @@ package menu
 import (
 	"strings"
 	"unicode/utf8"
+
+	"github.com/ViSiON-3/vision-3-bbs/internal/ansi"
 )
```
[the remark this answers is not in the document]
Comment (robbiew) [reply internal/menu/executor_oneliners_text.go:8]: Declining this one — `unicode/utf8` is still in use in that file, at lines 84, 85, 127, 168 and 183 (`DecodeRuneInString`, `RuneError`, `RuneCountInString` in `pipeCodeLenAt`, `truncateOnelinerPreservePipeCodes`, `truncatePipeCodedText` and `formatOnelinerDisplayName`). Only `truncateRunes` stopped using it, and the import is still required. `go build ./internal/menu/` passes at head, which it could not if the import were unused.

File: internal/menu/executor_oneliners_text_test.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package menu
+
+import "testing"
+
+// truncateRunes (oneliners) trims surrounding whitespace before measuring,
+// then appends a 3-rune "..." ellipsis when it cuts. These cases pin its
+// current behaviour before it is migrated to delegate to ansi.TruncateRunes.
```
[the remark this answers is not in the document]
Comment (robbiew) [reply internal/menu/executor_oneliners_text_test.go]: Fixed in aee3354 — the comment was written in the tests-first commit and described the migration as upcoming, which is stale once read from main. It now states what the test guards: behaviour the migration had to preserve exactly, and still must.

File: internal/usereditor/pubkey_dialog_test.go [hunk 3]
```diff
@@ -13,6 +13,38 @@ import (
 	"github.com/ViSiON-3/vision-3-bbs/internal/user"
 )
 
+// truncateRunes (pubkey dialog) has no ellipsis and no TrimSpace: it is a
+// plain rune-aware hard cut, used for the key-list comment column and error
+// text. These cases pin its current behaviour before it is migrated to
+// delegate to ansi.TruncateRunes.
```
[the remark this answers is not in the document]
Comment (robbiew) [reply internal/usereditor/pubkey_dialog_test.go]: Fixed in aee3354 — the comment was written in the tests-first commit and described the migration as upcoming, which is stale once read from main. It now states what the test guards: behaviour the migration had to preserve exactly, and still must.

File: internal/menu/executor_admin_users_test.go [hunk 4]
```diff
@@ -0,0 +1,36 @@
+package menu
+
+import "testing"
+
+// adminTruncate trims surrounding whitespace before measuring, then appends a
+// single-rune "…" ellipsis when it cuts. These cases pin its current
+// behaviour before it is migrated to delegate to ansi.TruncateRunes.
```
[the remark this answers is not in the document]
Comment (robbiew) [reply internal/menu/executor_admin_users_test.go]: Fixed in aee3354 — the comment was written in the tests-first commit and described the migration as upcoming, which is stale once read from main. It now states what the test guards: behaviour the migration had to preserve exactly, and still must.

File: internal/usereditor/view_test.go [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+import (
+	"testing"
+	"unicode/utf8"
+)
+
+// centerText (view.go) is currently byte-based (len(s) >= width, s[:width]).
+// Every caller today passes ASCII literals, so byte and rune counting agree —
+// these cases pin that observable behaviour before the helper is migrated to
+// delegate to ansi.Center/ansi.TruncateRunes as a genuine (if today
+// unreachable) correctness fix.
```
[the remark this answers is not in the document]
Comment (robbiew) [reply internal/usereditor/view_test.go]: Fixed in aee3354 — the comment was written in the tests-first commit and described the migration as upcoming, which is stale once read from main. It now states what the test guards: behaviour the migration had to preserve exactly, and still must.
